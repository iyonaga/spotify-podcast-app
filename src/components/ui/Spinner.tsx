import clsx from 'clsx';

interface Props {
  className?: string;
}

const Spinner: React.VFC<Props> = ({ className }) => (
  <div className={clsx('flex justify-center', className)}>
    <div className="inline-block w-8 h-8 rounded-full border-4 border-[#5d5f81] border-t-transparent animate-spin" />
  </div>
);

export default Spinner;
