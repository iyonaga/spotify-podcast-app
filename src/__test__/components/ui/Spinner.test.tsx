import { render } from '@testing-library/react';
import Spinner from '@/components/ui/Spinner';

describe('components/ui/Spinner.tsx', () => {
  test('スナップショットが一致する', () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
