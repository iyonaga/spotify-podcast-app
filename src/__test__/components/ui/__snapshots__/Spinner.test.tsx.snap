// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/Spinner.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center"
  >
    <div
      class="inline-block w-8 h-8 rounded-full border-4 border-[#5d5f81] border-t-transparent animate-spin"
    />
  </div>
</DocumentFragment>
`;
